### ISSUE TITLE
`throwAssertion` does not work on macOS

### ISSUE THREAD

**Issue by yuri-qualtie** [Quick/Nimble#1110]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

We run following spec for macOS platform:

```
                it("should NOT be called") {
                    expect(subject.removeAccounts()).to(throwAssertion())
                }
```
Note: subject.removeAccounts does `assertionFailure("should not be called")`

### What did you expect to happen?

Spec is passed.
BTW, we see this behavior in `Nimble 13.0.0`

### What actually happened instead?

Spec is crashed.
Console has message
```
Note: throwAssertion no longer works on tvOS or watchOS platforms when you use Nimble with Cocoapods.
      You will have to use Nimble with Swift Package Manager or Carthage.
```

It looks like `arch` or `canImport` is incorrectly resolved in `ThrowAssertion.swift`

### Environment

List the software versions you're using:

- Quick: *7.3.0*
- Nimble: *13.1.1*
- Xcode Version: *Version 15.0.1 (15A507)* (Open Xcode; In menubar: Xcode > About Xcode)
- Swift Version: *Default* (Open Xcode Preferences; Components > Toolchains. If none, use `Xcode Default`.)
- MacOS: *14.1.2*. 

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

- Cocoapods: *1.14.3* (Use `pod --version` in Terminal)

**Comment by svenmuennich** [comment]
I noticed the same behavior and narrowed it down to a missing dependency. https://github.com/Quick/Nimble/pull/1111 fixes this.

**Comment by younata** [comment]
This should be fixed in v13.1.2!
